velox/dwio/common/IntDecoder.cpp

```diff
@@ -2570,6 +2570,11 @@ void IntDecoder<isSigned>::decodeBitsLE(
     const char* bufferEnd,
     T* FOLLY_NONNULL result) {
   uint64_t mask = bits::lowMask(bitWidth);
+  if (bitWidth == 0) {
```

Comment:
This code is shared between Parquet and DWRF, right? If so, is this condition valid for DWRF? If not, should we throw if it happens?

Reply:
We should probably check `bitWidth` before calling this and throw if it is not valid in dwrf.  It is not easy to differentiate them inside the function.

Reply:
Looks like this method is called from a single place in a Parquet reader: velox/dwio/parquet/reader/RleDecoder.h. If that's the case, it is a bit confusing to have it in the "common" IntDecoder.

```
    super::decodeBitsLE(
        reinterpret_cast<const uint64_t*>(super::bufferStart),
        bitOffset_,
        folly::Range<const int32_t*>(rows + rowIndex, numRows),
        currentRow,
        bitWidth_,
        super::bufferEnd,
        reinterpret_cast<TIndex*>(values) + numValues);
```

Reply:
My guess is alpha will use that so it's put here for future proofing.